flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveScheduler.java

```diff
@@ -275,6 +275,10 @@ private static void ensureFullyPipelinedStreamingJob(JobGraph jobGraph)
                 "The adaptive scheduler only supports streaming jobs.");
 
         for (JobVertex vertex : jobGraph.getVertices()) {
+            Preconditions.checkState(
```

Comment:
this doesn't really belong into this method; it is unrelated to being a pipelined streaming job

Reply:
True, good catch. I actually planned to rename the method but missed it in the end.